crates/minecraft_packets/src/play/data/light_data.rs

```diff
@@ -18,36 +19,84 @@ pub struct Light {
 }
 
 impl LightData {
-    pub fn new_with_level(light_level: u8) -> Self {
-        const NUM_SECTIONS_IN_WORLD: u32 = 24;
-        let light_level = light_level.clamp(0, 15);
+    pub fn from_light_data(
+        sky_light_sections: &[LightSection],
+        block_light_sections: &[LightSection],
+        dimension: Dimension,
+    ) -> Self {
+        let world_section_count = dimension.height() / 16;
+        let total_light_sections = (world_section_count + 2) as u32;
 
-        if light_level == 0 {
-            return Self::default();
+        let all_sections_mask_val = (1u64 << total_light_sections) - 1;
+        let all_sections_mask = BitSet::new(vec![all_sections_mask_val as i64]);
+
+        let mut sky_light_arrays = Vec::with_capacity(total_light_sections as usize);
+        sky_light_arrays.push(Light {
+            block_light_array: LengthPaddedVec::new(vec![0xFFu8 as i8; 2048]),
+        });
+        for section in sky_light_sections {
+            sky_light_arrays.push(Light {
```

Comment:
You could create a constructor for the Light struct, that accepts a regular Vec (not a LengthPaddedVec.)